File: cafeland-club-frontend/src/app/features/miembros/services/miembro.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import {
  EstadoMiembroRequest,
  Miembro,
  MiembroActualizarRequest,
  MiembroCrearRequest,
} from '../models/miembro.model';

@Injectable({ providedIn: 'root' })
export class MiembroService {
  private readonly http = inject(HttpClient);
  private readonly apiUrl = '/api/miembros';

  listarTodos(): Observable<Miembro[]> {
    return this.http.get<Miembro[]>(this.apiUrl);
  }

  listarActivos(): Observable<Miembro[]> {
    return this.http.get<Miembro[]>(`${this.apiUrl}/activos`);
  }

  registrar(request: MiembroCrearRequest): Observable<Miembro> {
    return this.http.post<Miembro>(this.apiUrl, request);
  }

  actualizar(id: number, request: MiembroActualizarRequest): Observable<Miembro> {
    return this.http.put<Miembro>(`${this.apiUrl}/${id}`, request);
  }

  cambiarEstado(id: number, request: EstadoMiembroRequest): Observable<Miembro> {
    return this.http.put<Miembro>(`${this.apiUrl}/${id}/estado`, request);
  }
}

File: cafeland-club-backend/src/main/java/com/cafeland/club/service/MiembroService.java
package com.cafeland.club.service;

import com.cafeland.club.dto.miembro.MiembroActualizarRequest;
import com.cafeland.club.dto.miembro.MiembroCrearRequest;
import com.cafeland.club.dto.miembro.MiembroResponse;
import com.cafeland.club.entity.EstadoMiembro;

import java.util.List;
import java.util.Optional;

public interface MiembroService {

    List<MiembroResponse> listarTodos();

    Optional<MiembroResponse> obtenerPorId(Long id);

    List<MiembroResponse> listarPorEstado(EstadoMiembro estado);

    MiembroResponse registrar(MiembroCrearRequest request);

    Optional<MiembroResponse> actualizar(Long id, MiembroActualizarRequest request);

    Optional<MiembroResponse> cambiarEstado(Long id, EstadoMiembro estado);
}
